src/commands/schedule.ts

```diff
@@ -1,38 +1,49 @@
 import type { Command } from "./type";
+import { GuildChannel, TextChannel } from "discord.js";
 import { scheduleJob } from "node-schedule";
+import { validate, days } from "./helpers/scheduleValidators";
 
-const days: Record<string, number> = {
-  Sunday: 0,
-  Monday: 1,
-  Tuesday: 2,
-  Wednesday: 3,
-  Thursday: 4,
-  Friday: 5,
-  Saturday: 6,
-};
+const isTextChannel = (channel: GuildChannel): channel is TextChannel => channel.type === "text";
 
 export const schedule: Command = {
   name: "schedule",
   description: "Schedule reoccuring messages",
-  definition: "schedule :day :time *",
-  async execute(command, { day, time, message }): Promise<void> {
-    if (day in days === false) {
-      await command.reply("Invalid day argument. Spell full name of day");
+  definition: "schedule :day :time :channelStr *",
+  async execute(command, { day, time, channelStr, message }): Promise<void> {
+    if (!validate.day(day)) {
+      await command.reply("Invalid day argument. Day must be spelt in full");
+      return;
+    }
+
+    if (!validate.time(time)) {
+      await command.reply("Invalid time argument. Must use 24 hour time seperated by :");
       return;
     }
 
-    const timeRegex = /^([01]?[0-9]|2[0-3]):[0-5][0-9]$/;
-    if (!timeRegex.exec(time)) {
-      await command.reply("Invalid time argument (24 hour time eg 13:30)");
+    if (!validate.channel(command.channel, "text")) {
```

Comment:
Couldn't this just reuse `isTextChannel` function? That would help with the type guards I think.